infrastructure/code/bundle.go

```diff
@@ -94,6 +95,27 @@ func getIssueLangAndRuleId(issue snyk.Issue) (string, string, bool) {
 	return ruleIdSplit[0], ruleIdSplit[1], true
 }
 
+// isAutofixSupportedForExtension shall call `GetFilters` if necessary and cache the results
+// in the `codeSettings` singleton.
+func (b *Bundle) isAutofixSupportedForExtension(ctx context.Context, file string) bool {
+	if getCodeSettings().autofixExtensions == nil {
+		// query
+		_, _, autofixExts, err := b.SnykCode.GetFilters(ctx)
```

Comment:
We call and cache `GetFilters` in [BundleUploader](https://github.com/snyk/snyk-ls/blob/6d2a941c53e0effd4d5cd1c9e7f43143c506000f/infrastructure/code/bundle_uploader.go#L119) already. Can we save a network call and unify caching between there and here?

We can introduce something like "Filters" type shared for the whole scanner.

Reply:
I considered that, but 
- the autofix supported extensions logically do not belong to the bundle uploading. I understand why `supportedExtension` are needed to bundle uploader at the bundle, but the autofix supported extensions come into play later so they are called at most once at the codeaction creation time (and if the autofix is enabled)
- to address your comment I could have made autofix caching at the call site by the `bundleUploader.isSupporte` -- but I would cosider that logically much less sound (why does `isSupported` do anything for autofix caching?)
- the remaining argument about saving a call: I don't think that one literally one call (which is cached on the side of deeproxy) while there are orders of magnitude more calls when e.g. polling a long analysis results is something to worry about

Reply:
Agree 👍 There's no harm in issuing this additional call. It can be little confusing why this state isn't shared already, if it's the same endpoint architecturally. But it's really a minor thing and pragmatically not worth of investment here, just wanted to flag :)